src/pseudopeople/interface.py

```diff
@@ -173,14 +173,21 @@ def generate_decennial_census(
     verbose: bool = False,
 ) -> pd.DataFrame:
     """
-    Generates noised decennial census data from un-noised data.
-
-    :param source: The root directory of clean data input which needs to be noised
-    :param seed: An integer seed for randomness
-    :param config: (optional) A path to a configuration YAML file or a dictionary to override the default configuration
-    :param year: The year from the data to noise
-    :param verbose: Log with verbosity if True. Default is False.
-    :return: A pd.DataFrame of noised census data
+    Generates noised data from the pseudopeople decennial census dataset. This
+    noised data represents simulated responses to the US Census Bureau's Census
+    of Population and Housing.
+
+    :param source: The root directory containing the datasets to be noised. Defaults to the pseudopeople sample datasets directory.
+    :param seed: An integer seed for randomness.
+    :param config: An optional override to the default configuration. Can be a path to a configuration YAML file or a dictionary.
+    :param year: The year (format YYYY) to noise in the dataset. Must be a decennial year (eg 2020, 2030, 2040). Will return an empty pd.DataFrame if there is no data with this year.
+    :param verbose: Log with verbosity if True.
+    :return: A pd.DataFrame of noised decennial census data.
+    :raises ConfigurationError: An incorrect config is provided.
+    :raises TypeError: More than one file type is found in the source directory provided.
```

Comment:
Let's create a custom error for these cases. Maybe `DataSourceError`? Then you can just say 
```
:raises DataSourceError: An incorrect data source is provided.
```